src/RA_httpthread.cpp

```diff
@@ -314,22 +309,20 @@ BOOL RAWeb::DoBlockingRequest(RequestType nType, const PostArgs& PostData, Docum
     return FALSE;
 }
 
-BOOL RAWeb::DoBlockingRequest(RequestType nType, const PostArgs& PostData, std::string& Response)
+BOOL RAWeb::DoBlockingRequest(ra::RequestType nType, const PostArgs& PostData, std::string& Response)
 {
-    PostArgs args = PostData;               //  Take a copy
-    args['r'] = RequestTypeToPost[nType];   //  Embed request type
-
-    switch (nType)
-    {
-        case RequestUserPic:
-            return DoBlockingHttpGet(std::string("UserPic/" + PostData.at('u') + ".png"), Response, false); //  UserPic needs migrating to S3...
-        case RequestBadge:
-            return DoBlockingHttpGet(std::string("Badge/" + PostData.at('b') + ".png"), Response, true);
-        case RequestLogin:
-            return DoBlockingHttpPost("login_app.php", PostArgsToString(args), Response);
-        default:
-            return DoBlockingHttpPost("dorequest.php", PostArgsToString(args), Response);
-    }
+    PostArgs args = PostData; //  Take a copy
+    args['r'] = ra::RequestTypeToPost.at(ra::etoi(nType)); //  Embed request type
+    
+    if( nType == ra::RequestType::Login) 
+        return DoBlockingHttpPost("login_app.php", PostArgsToString(args), Response);
```

Comment:
Falling through caused problems, so it's an if else instead. there's a few other switches that were changed as well.